### PR TITLE
Add a page that shows cmos rates over time for specified c/c/c

### PR DESCRIPTION
**Pull request by tannerbk**
Adds a new page that shows the cmos rate vs run number for a select crate/card/channel.
rates_plot

### CODE REVIEW

File: minard/templates/check_rates_history.html [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                    <th>Mean</th> 
+                    <th>Minimum</th> 
+                    <th>Maximum</th>
+                    <th>VThr</th> 
+                </tr>
+                <tr>
+                    <td> {{ stats[2] }} &plusmn; {{ stats[3] }} </td>
+                    <td> {{ stats[1] }} </td>
+                    <td> {{ stats[0] }} </td>
+                    <td> {{ discriminator_threshold['vthr'][channel] -                                                      discriminator_threshold['zero_disc'][channel] }} </td>
```
Comment (tlatorre-uchicago) [inline minard/templates/check_rates_history.html]: Is this a typo or is this some jinja syntax that I'm not aware of?

Comment (tlatorre-uchicago) [inline minard/templates/check_rates_history.html]: ah nm. it's just off of the page on github

File: minard/static/js/scatter_plot_rates.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    var last_run = data[0][0];
+    var first_run = data[data_length-1][0]
+    var len = (last_run - first_run)/data.length;
+    var run_offset = 0;
+    var start_offset = 103214;
+    var start = start_offset - run_offset;
+    var end = d3.max(data, function(d) { return d[0]; }) + 1 - run_offset;
+    
+    var margin = {top: 20, right: 15, bottom: 60, left: 260}
+        , width = 1600 - margin.left - margin.right
```
Comment (tlatorre-uchicago) [inline minard/static/js/scatter_plot_rates.js]: For the width here, you can grab the screen width by selecting the div that it's going to be added to and getting the width, i.e.:

```
width = $("#main").width();
```

Currently, it shows up really big on my browser:


check_rates_history

File: minard/polling.py [hunk 3]
```diff
@@ -27,6 +26,43 @@ def polling_runs():
...
 
+def polling_history(crate, slot, channel):
+    '''
+    Return a list of form [[run number, cmos rate]]
+    for all runs with cmos data polling 
+    '''
+
+    conn = engine.connect()
+
+    result = conn.execute("SELECT run, cmos_rate from cmos WHERE crate = %i \
```
Comment (tlatorre-uchicago) [inline minard/polling.py]: Not super critical, but it's better to use %s and then pass the crate, card, channel as a tuple. See here: http://initd.org/psycopg/docs/usage.html#passing-parameters-to-sql-queries.

File: minard/polling.py [hunk 4]
```diff
@@ -27,6 +26,46 @@ def polling_runs():
...
+    Return a list of form [[run number, cmos rate]].
+    for all runs with cmos data polling. Also returns
+    a list which included statistics on the cmos data.
+    '''
+
+    conn = engine.connect()
+
+    result = conn.execute("SELECT run, cmos_rate from cmos WHERE crate = %s \
+                           AND slot = %s AND channel = %s ORDER by run DESC" \
+                           % (crate, slot, channel))
```
Comment (tlatorre-uchicago) [inline minard/polling.py]: The args need to be passed as an argument, i.e.

```
result = conn.execute("SELECT run, cmos_rate FROM cmos WHERE crate = %s AND slot = %s AND channel = %s ORDER BY run DESC", (crate, slot, channel))
```

**Comment by tannerbk** [comment]
OK with the changes this is what the page should look like now:
changes

File: minard/static/js/scatter_plot_rates.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    var last_run = data[0][0];
+    var first_run = data[data_length-1][0]
+    var len = Math.floor((last_run - first_run)/data.length);
+    var run_offset = 0;
+    var start_offset = 103214;
+    var start = start_offset - run_offset;
+    var end = d3.max(data, function(d) { return d[0]; }) + 1 - run_offset; 
+
+    var margin = {top: 10, right: 80, bottom: 80, left: 120}
+        ,width = $("#main").width()
```
Comment (tlatorre-uchicago) [inline minard/static/js/scatter_plot_rates.js]: I think you want to subtract the margins from this so it fits on one page (https://github.com/snoplus/minard/blob/master/minard/static/js/histogram.js#L25).

File: minard/templates/check_rates_history.html [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+                    <td> {{ discriminator_threshold['vthr'][channel] - discriminator_threshold['zero_disc'][channel] }} </td>
+                </tr>
+            </table>
+            {% endif %}
+        </div>
+    </div>
+    <div class="row">
+        <div class="col-md-12" id="main">
+            {% if data %}
+                <script type="text/javascript" src="js/scatter_plot_rates.js"></script>
```
Comment (tlatorre-uchicago) [inline minard/templates/check_rates_history.html]: Is this meant to be here? You include it at the bottom. If it is meant to be here, it should be:

```javascript
<script src="{{ url_for('static', filename='js/scatter_lot_rates.js') }}"></script>
```

**Comment by tlatorre-uchicago** [comment]
Seems to work now, but there is a vertical line to the left of the plot.

check_rates_history_vertical_line

One thing that may be easier than trying to use d3 to do the plot is to use the metrics graphics library. You can see how it's used here: https://github.com/snoplus/minard/blob/master/minard/static/js/graph.js. I think they also offer a way to do baselines. See here: https://www.metricsgraphicsjs.org/examples.htm#annotations under the baselines example.

**Pull request merged**